src/global_distrib/mod_global_distrib_mapping_redis.erl

```diff
@@ -279,15 +285,23 @@ set_endpoints(Endpoints) ->
     refresh_set(EndpointsKey, BinEndpoints).
 
 -spec endpoint_to_binary(mod_global_distrib_utils:endpoint()) -> binary().
-endpoint_to_binary({IpAddr, Port}) ->
-    iolist_to_binary([inet:ntoa(IpAddr), "#", integer_to_binary(Port)]).
+endpoint_to_binary({IpAddr, Port}) when is_tuple(IpAddr) ->
+    iolist_to_binary([inet:ntoa(IpAddr), "#", integer_to_binary(Port)]);
+endpoint_to_binary({Domain, Port}) when is_list(Domain) ->
+    iolist_to_binary([list_to_binary(Domain), "#", integer_to_binary(Port)]).
 
 -spec binary_to_endpoint(binary()) -> mod_global_distrib_utils:endpoint().
 binary_to_endpoint(Bin) ->
     [Addr, BinPort] = binary:split(Bin, <<"#">>),
-    {ok, IpAddr} = inet:parse_address(binary_to_list(Addr)),
+    IpAddrOrDomain = case mod_global_distrib_utils:is_domain(Addr) of
```

Comment:
`is_domain` does exactly the same thing as line 298. Can we move it here and match on the result?